commitizen-tools/commitizen
# feat(commit): add retry_after_failure config option and --no-retry flag

## Description

This pull request resolves #732 by implementing the requested feature.

## Checklist


## Expected behavior

See https://github.com/commitizen-tools/commitizen/issues/732#issue-1689976468.

## Steps to Test This Pull Request

1. Set up a git repository with a git hook that always fails
```
git init test
cd test
echo '#!/usr/bin/env bash\necho "pre-commit fail"\nfalse' > .git/hooks/pre-commit 
chmod +x .git/hooks/pre-commit
```
2. Enable `retry_after_failure`
```
echo '[tool.commitizen]\nretry_after_failure=true' > .cz.toml
```
3. Try to create a commit (should fail)
```
git add .
cz commit
```
4. Retry (still fails but reuses the previous message)
```
cz commit
```
5. Force new message prompt (still fails but uses new message)
```
cz commit --no-retry
```

## Additional context
